File: core/MaximizeTheConfusionOfAnExam/MaximizeTheConfusionOfAnExam.java
package com.dbc;

public class MaximizeTheConfusionOfAnExam {
    public int maxConsecutiveAnswers(String answerKey, int k) {
        int left = 0, right = 0, res = 0;
        int fCount = 0, tCount = 0;
        while (right < answerKey.length()) {
            while (right < answerKey.length() && (fCount <= k || tCount <= k)) {
                if (answerKey.charAt(right) == 'T') tCount++;
                else fCount++;
                right++;
            }

            res = Math.max(res, (fCount <= k || tCount <= k) ? right - left : right - left - 1);

            while (right < answerKey.length() && left < right && (fCount > k && tCount > k)) {
                if (answerKey.charAt(left) == 'T') tCount--;
                else fCount--;
                left++;
            }
        }
        res = Math.max(res, (fCount <= k || tCount <= k) ? right - left : right - left - 1);
        return res;
    }
}

File: core/MaximizeTheConfusionOfAnExam/MaximizeTheConfusionOfAnExam.py
def maxConsecutiveAnswers(answerKey: str, k: int) -> int:
    length = len(answerKey)
    left, right = 0, 0
    res, t_count, f_count = 0, 0, 0
    while right < length:
        while right < length and (t_count <= k or f_count <= k):
            if answerKey[right] == "T":
                t_count += 1
            else:
                f_count += 1
            right += 1

        res = max(res, right - left if t_count <= k or f_count <= k else right - left - 1)

        while right < length and left < right and (t_count > k and f_count > k):
            if answerKey[left] == "T":
                t_count -= 1
            else:
                f_count -= 1
            left += 1

    res = max(res, right - left if t_count <= k or f_count <= k else right - left - 1)

    return res

if __name__ == '__main__':
    print(maxConsecutiveAnswers(answerKey = "FFFTTFTTFT", k = 3))

